### ISSUE TITLE
Bump fuels* versions to the v0.24.0

### ISSUE THREAD

**Issue by kayagokalp** [FuelLabs/forc-wallet#44]
Realized while testing https://github.com/FuelLabs/fuels-rs/pull/578 locally with `forc-wallet` (#28). Since we need to make some changes (from `Wallet` to `WalletUnlocked`) I am planning to tackle this in a separate PR (rather than with #28).

**Comment by kayagokalp** [comment]
This will unblock #28. But first I need to wait for a release since https://github.com/FuelLabs/fuels-rs/pull/578 is merged now.
